hudi-client/src/main/java/org/apache/hudi/index/hbase/HBaseIndex.java

```diff
@@ -177,13 +176,11 @@ private Get generateStatement(String key) throws IOException {
   }
 
   private boolean checkIfValidCommit(HoodieTableMetaClient metaClient, String commitTs) {
-    HoodieTimeline commitTimeline = metaClient.getActiveTimeline().filterCompletedInstants();
+    HoodieTimeline commitTimeline = metaClient.getCommitsTimeline().filterCompletedInstants();
```

Comment:
Can you please change this to -> `metaClient.getActiveTimeline().getCommitsTimeline()`. For context, the getActiveTimeline() loads the timeline once and then caches it. The getCommitsTimeline() call will not force this, in this case it's working by coincidence because the metaClient passed already has the loaded timeline from the caller but I would want to avoid having that assumption. 
@rmpifer

Reply:
@n3nash It looks like HoodieTableMetaClient.getCommitsTimeline internally calls getActiveTimeline()
https://github.com/rmpifer/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java#L478

Reply:
okay, great, thanks for checking.